static/events.js

```diff
@@ -34,10 +34,24 @@
           window.parent.postMessage({ type: 'a-plus-refresh-stats' }, "*");
         }
         if (cb) {
-          cb(response.content);
+          var error = null;
+          if (response.status !== 'OK') {
+            error = {
+              'status': response.status,
+              'error': response.error, // error message (string)
+            };
+          }
+          cb(response.content, error);
         }
-      }).fail(function(jqXHR, text) {
+      }).fail(function(jqXHR, text, errorThrown) {
         console.log(jqXHR, text);
+        if (cb) {
+          var error = {
+            'status': 'ERROR',
+            'error': errorThrown.toString(),
+          };
+          cb('', error);
```

Comment:
Sure, it could be. The first argument is probably not used when the error is not null. I used an empty string for the first argument since normally the content is a string, but it might as well be null.